internal/util/password/scramsha256_test.go

```diff
@@ -0,0 +1,230 @@
+			"iterationCount", int32(4096),
+			"salt", "c2FsdFNBTFRzYWx0U0FMVHNhbHRTQUxUc2FsdFNBTFRzYWx0",
+			"serverKey", "4QTMss7Dzi+pk8C8cyql++OaWqI0y/FyhXJI7W9acHI=",
+		)),
+	},
+	{
+		params: scramSHA256Params{
+			iterationCount: 4096,
+			saltLen:        5,
+		},
+		username: "username",
+		password: "pass\x00word",
+		salt:     []byte("sa\x00lt"),
+		err:      "prohibited character",
+	},
+	{
+		params: scramSHA256Params{
+			iterationCount: 1,
+			saltLen:        4,
+		},
+		username: "username",
+		password: "passwd",
+		salt:     []byte("salt"),
+		want: must.NotFail(types.NewDocument(
+			"storedKey", "dcwmgrDYICpRpKjwHLKxZ/21/go62U106s5V4i9v+Q8=",
+			"iterationCount", int32(1),
+			"salt", "c2FsdA==",
+			"serverKey", "qFes4m5Z84MaC2hSJqCR2e/FBz7goMVu/RTRNnb5Fj0=",
+		)),
+	},
+	{
+		params: scramSHA256Params{
+			iterationCount: 80000,
+			saltLen:        4,
+		},
+		username: "username",
+		password: "Password",
+		salt:     []byte("NaCl"),
+		want: must.NotFail(types.NewDocument(
+			"storedKey", "EI8wmB+eWKGZ8k+dln75YQnX8lj+MBoG6+eH9AzR1e4=",
+			"iterationCount", int32(80000),
+			"salt", "TmFDbA==",
+			"serverKey", "Wbwo6JsaJrZ/1Bf7F+45jY2VURuezLXxADxUuzWdZ/4=",
+		)),
+	},
+	{
+		params: scramSHA256Params{
+			iterationCount: 4096,
+			saltLen:        5,
+		},
+		username: "username",
+		password: "Password",
+		salt:     []byte("sa\x00lt"),
+		want: must.NotFail(types.NewDocument(
+			"storedKey", "BHJbwIZ9YCvb+dFApByBLe4jR7gquvBm3kPApnWCylk=",
+			"iterationCount", int32(4096),
+			"salt", "c2EAbHQ=",
+			"serverKey", "qpmMln7z50yoT66R1lT55pUvBxu8BqZU4X8dRp38NWo=",
+		)),
+	},
+}
+
+func TestSCRAMSHA256(t *testing.T) {
+	for i, tc := range scramSHA256TestCases {
+		i, tc := i, tc
+		t.Run(fmt.Sprintf("%d", i), func(t *testing.T) {
+			t.Parallel()
+
+			doc, err := scramSHA256HashParams(tc.username, tc.password, tc.salt, &tc.params)
+
+			if tc.err != "" {
+				assert.Contains(t, err.Error(), tc.err)
+				return
+			}
+
+			require.NoError(t, err)
+			testutil.AssertEqual(t, tc.want, doc)
+
+			scramServer, err := scram.SHA256.NewServer(func(username string) (scram.StoredCredentials, error) {
+				return scram.StoredCredentials{
+					KeyFactors: scram.KeyFactors{
+						Salt:  string(tc.salt),
+						Iters: tc.params.iterationCount,
+					},
+					StoredKey: []byte(must.NotFail(doc.Get("storedKey")).(string)),
+					ServerKey: []byte(must.NotFail(doc.Get("serverKey")).(string)),
+				}, nil
+			})
+			must.NoError(err)
+
+			// Check if the generated authentication is valid by simulating a conversation.
+			conv := scramServer.NewConversation()
+
+			client, err := scram.SHA256.NewClient(tc.username, tc.password, "")
+			require.NoError(t, err)
+
+			resp, err := client.NewConversation().Step("")
+			require.NoError(t, err)
+
+			resp, err = conv.Step(resp)
+			require.NoError(t, err)
+			assert.NotEmpty(t, resp)
+
+			_, err = conv.Step("wrong")
+			assert.NotNil(t, err)
```

Comment:
Or `assert.EqualError()` for comparing specific error. 
```suggestion
			assert.Error(t, err)
```